api/v2/checluster_types.go

```diff
@@ -0,0 +1,637 @@
+	// In a conjunction with labels it creates a route, which is served by a non-default Ingress controller.
+	// For Kubernetes cluster it contains a global ingress domain. This MUST be explicitly specified: there are no defaults.
+	// +optional
+	Domain string `json:"domain,omitempty"`
+	// Public hostname of the installed Che server. When value is omitted, the value it will be automatically set by the Operator.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
+	// Name of a secret that will be used to setup ingress TLS termination when.
+	// When the field is empty string, the default cluster certificate will be used.
+	// The secret must have `app.kubernetes.io/part-of=che.eclipse.org` label.
+	// +optional
+	TlsSecretName string `json:"tlsSecretName,omitempty"`
+	// Authentication settings.
+	// +optional
+	Auth Auth `json:"auth"`
+}
+
+// Container registry configuration.
+// +k8s:openapi-gen=true
+type CheClusterContainerRegistry struct {
+	// Optional hostname, or URL, to an alternate container registry to pull images from.
+	// This value overrides the container registry hostname defined in all the default container images involved in a Che deployment.
+	// This is particularly useful to install Che in a restricted environment.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
+	// Optional repository name of an alternate container registry to pull images from.
+	// This value overrides the container registry organization defined in all the default container images involved in a Che deployment.
+	// This is particularly useful to install Eclipse Che in a restricted environment.
+	// +optional
+	Organization string `json:"organization,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// General configuration settings related to the Che server.
+type CheServer struct {
+	// Deployment override options.
+	// +optional
+	Deployment Deployment `json:"deployment,omitempty"`
+	// Log level for the Che server: `INFO` or `DEBUG`.
+	// +optional
+	// +kubebuilder:default:="INFO"
+	LogLevel string `json:"logLevel,omitempty"`
+	// Enables the debug mode for Che server.
+	// +optional
+	Debug bool `json:"debug,omitempty"`
+	// ClusterRoles that will be assigned to Che ServiceAccount.
+	// Defaults roles are:
+	// - `<che-namespace>-cheworkspaces-namespaces-clusterrole`
+	// - `<che-namespace>-cheworkspaces-clusterrole`
+	// - `<che-namespace>-cheworkspaces-devworkspace-clusterrole`
+	// where <che-namespace> is the namespace where the CheCluster CRD is created.
+	// Each role must have `app.kubernetes.io/part-of=che.eclipse.org` label.
+	// Be aware that the Che Operator has to already have all permissions in these ClusterRoles to grant them.
+	// +optional
+	ClusterRoles []string `json:"clusterRoles,omitempty"`
+	// Proxy server settings.
+	// +optional
+	Proxy Proxy `json:"proxy"`
```

Comment:
Consider specifying how this field should be used -- the Operator will automatically read proxy configs on OpenShift.